# Import library

In [8]:
import numpy as np
import pandas as pd
import seaborn as sns

# Import Data

In [36]:
data = pd.read_csv('data/data.csv', encoding= 'latin-1')
df_kamusalay = pd.read_csv('data/new_kamusalay.csv', names = ['aslinya','pengganti'], encoding='latin-1')
df_abusivetext = pd.read_csv('data/abusive.csv', encoding='latin1')

In [10]:
data.head()

Tweet  HS  Abusive  \
0  - disaat semua cowok berusaha melacak perhatia...   1        1   
1  RT USER: USER siapa yang telat ngasih tau elu?...   0        1   
2  41. Kadang aku berfikir, kenapa aku tetap perc...   0        0   
3  USER USER AKU ITU AKU\n\nKU TAU MATAMU SIPIT T...   0        0   
4  USER USER Kaum cebong kapir udah keliatan dong...   1        1   

   HS_Individual  HS_Group  HS_Religion  HS_Race  HS_Physical  HS_Gender  \
0              1         0            0        0            0          0   
1              0         0            0        0            0          0   
2              0         0            0        0            0          0   
3              0         0            0        0            0          0   
4              0         1            1        0            0          0   

   HS_Other  HS_Weak  HS_Moderate  HS_Strong  
0         1        1            0          0  
1         0        0            0          0  
2         0        0            0          0  
3         0        0            0          0  
4         0        0            1          0

In [11]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13169 entries, 0 to 13168
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Tweet          13169 non-null  object
 1   HS             13169 non-null  int64 
 2   Abusive        13169 non-null  int64 
 3   HS_Individual  13169 non-null  int64 
 4   HS_Group       13169 non-null  int64 
 5   HS_Religion    13169 non-null  int64 
 6   HS_Race        13169 non-null  int64 
 7   HS_Physical    13169 non-null  int64 
 8   HS_Gender      13169 non-null  int64 
 9   HS_Other       13169 non-null  int64 
 10  HS_Weak        13169 non-null  int64 
 11  HS_Moderate    13169 non-null  int64 
 12  HS_Strong      13169 non-null  int64 
dtypes: int64(12), object(1)
memory usage: 1.3+ MB


Data CSV terdiri dari 13 Kolom, dan memiliki 10 level Hatespeech. Pada data CSV memiliki 13,169 rows tanpa ada data null. Tipe data pada data CSV adalah 1 kolom data string dan 12 kolom data integer.

In [16]:
data_abusive = data['Abusive'] == 1
data_abusive.sum()

5043

In [14]:
data_HS = data['HS'] == 1
data_HS.sum()

5561

In [17]:
data_HS_individual = data['HS_Individual'] == 1
data_HS_individual.sum()

3575

In [18]:
data_HS_group = data['HS_Group'] == 1
data_HS_group.sum()

1986

In [19]:
data_HS_Religion = data['HS_Religion'] == 1
data_HS_Religion.sum()

793

In [20]:
data_HS_Race = data['HS_Race'] == 1
data_HS_Race.sum()

566

In [21]:
data_HS_Physical = data['HS_Physical'] == 1
data_HS_Physical.sum()

323

In [22]:
data_HS_Gender = data['HS_Gender'] == 1
data_HS_Gender.sum()

306

In [23]:
data_HS_Other = data['HS_Other'] == 1
data_HS_Other.sum()

3740

In [24]:
data_HS_Weak = data['HS_Weak'] == 1
data_HS_Weak.sum()

3383

In [25]:
data_HS_Moderate = data['HS_Moderate'] == 1
data_HS_Moderate.sum()

1705

In [26]:
data_HS_Strong = data['HS_Strong'] == 1
data_HS_Strong.sum()

473

In [38]:
df_kamusalay.head()

aslinya                 pengganti
0  anakjakartaasikasik  anak jakarta asyik asyik
1         pakcikdahtua         pak cik sudah tua
2       pakcikmudalagi         pak cik muda lagi
3          t3tapjokowi              tetap jokowi
4                   3x                 tiga kali

In [33]:
df_kamusalay.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15166 entries, 0 to 15165
Data columns (total 2 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   anakjakartaasikasik       15166 non-null  object
 1   anak jakarta asyik asyik  15166 non-null  object
dtypes: object(2)
memory usage: 237.1+ KB


In [7]:
df_abusivetext.head()

ABUSIVE
0     alay
1    ampas
2     buta
3  keparat
4   anjing

In [34]:
df_abusivetext.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 125 entries, 0 to 124
Data columns (total 1 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   ABUSIVE  125 non-null    object
dtypes: object(1)
memory usage: 1.1+ KB


# Data Cleansing

In [39]:
alay_dict_map = dict(zip(df_kamusalay['aslinya'], df_kamusalay['pengganti']))


In [40]:
alay_dict_map

{'anakjakartaasikasik': 'anak jakarta asyik asyik',
 'pakcikdahtua': 'pak cik sudah tua',
 'pakcikmudalagi': 'pak cik muda lagi',
 't3tapjokowi': 'tetap jokowi',
 '3x': 'tiga kali',
 'aamiin': 'amin',
 'aamiinn': 'amin',
 'aamin': 'amin',
 'aammiin': 'amin',
 'abis': 'habis',
 'abisin': 'habiskan',
 'acau': 'kacau',
 'achok': 'ahok',
 'ad': 'ada',
 'adek': 'adik',
 'adl': 'adalah',
 'adlah': 'adalah',
 'adlh': 'adalah',
 'ado': 'ada',
 'aduhh': 'aduh',
 'aer': 'air',
 'afdol': 'afdal',
 'agamaataualqur': 'agama alquran',
 'agm': 'agama',
 'agma': 'agama',
 'ahaha': 'haha',
 'ahahaha': 'haha',
 'ahehehehe': 'hehe',
 'ahir': 'akhir',
 'ahirnya': 'akhirnya',
 'ahk': 'ahok',
 'ahlamdulillah': 'alhamdulillah',
 'ahli2': 'para ahli',
 'ahlusunnah': 'ahlus sunah',
 'ahmaddani': 'ahmad dhani',
 'aho': 'ahok',
 'ahoax': 'ahok',
 'ahoaxx': 'ahok',
 'ahog': 'ahok',
 'ahokataudjarot': 'ahok djarot',
 'ahokbebanijokowi': 'ahok beban jokowi',
 'ahokbtp': 'ahok basuki tjahaja purnama',
 'ahokditolakw

In [41]:
def normalize_alay(text):
    return ' '.join([alay_dict_map[word] if word in alay_dict_map else word for word in text.split(' ')])
